controller/instance_manager_controller.go

```diff
@@ -955,17 +981,33 @@ func (imc *InstanceManagerController) createReplicaManagerPodSpec(im *longhorn.I
 		return nil, err
 	}
 
+	secretIsOptional := true
 	podSpec.ObjectMeta.Labels = types.GetInstanceManagerLabels(imc.controllerID, im.Spec.Image, longhorn.InstanceManagerTypeReplica)
 	podSpec.Spec.Containers[0].Name = "replica-manager"
 	podSpec.Spec.Containers[0].Args = []string{
 		"longhorn-instance-manager", "--debug", "daemon", "--listen", "0.0.0.0:8500",
 	}
+	podSpec.Spec.Containers[0].Env = []v1.EnvVar{
+		{
+			Name:  "TLS_DIR",
+			Value: types.TLSDirectoryInContainer,
+		},
+	}
 	podSpec.Spec.Containers[0].VolumeMounts = []v1.VolumeMount{
 		{
 			MountPath:        "/host",
 			Name:             "host",
 			MountPropagation: &hostToContainer,
 		},
+		{
+			MountPath:        types.EngineBinaryDirectoryInContainer,
+			Name:             "engine-binaries",
+			MountPropagation: &hostToContainer,
+		},
```

Comment:
Why replica IM pods need to access engine binaries?

Reply:
They already do access this via the root host mount, I would like to unify the two im pods to only refer to `engine-binaries` instead of doing the indirect access via the root host mount.

This will allow for removal of one of the white list entries and make the pathing between im-e and im-r identical for engine calls.

But it's not a requirement and can be removed for the moment.

Reply:
If you want to unify the binary path, you may need to update this as well: https://github.com/longhorn/longhorn-manager/blob/ee7a757d554ed57b4bd2487e80dba880e551333f/engineapi/instance_manager.go#L138

Reply:
I removed this for now from this PR, since we can do the full change over in the future.
Better to have all changes be contained in that PR.